tests/test_default_values.py

```diff
@@ -0,0 +1,164 @@
+in all copies or substantial portions of the Software.
+"""
+
+import unittest
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+from .data import car
+
+
+class TestDefaultValues(unittest.TestCase):
+    def test_return_empty_containers(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+
+        # Unspecified containers are defaulted to empty container of their type
+        self.assertEqual(config_suite.snapshot.owner, {})
+        self.assertEqual(config_suite.snapshot.color, {})
+        self.assertEqual(config_suite.snapshot.incidents, [])
+
+    def test_default_values(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+        self.assertTrue(config_suite.valid)
+
+        # Named dict top level
+        self.assertEqual(config_suite.snapshot.tire.dimension, 15)
+        self.assertEqual(config_suite.snapshot.tire.rim, "green")
+
+        # Named dict within dict
+        extracted_owner = {key: value for key, value in config_suite.snapshot.owner}
+        self.assertEqual(extracted_owner["first entry"].location, "Svalbard")
+        self.assertEqual(extracted_owner["first entry"].name, "Johan")
+        self.assertEqual(extracted_owner["second entry"].location, "Earth")
+        self.assertEqual(extracted_owner["second entry"].name, "Svein")
+
+        # Named dict within list
+        self.assertEqual(config_suite.snapshot.incidents[0].location, "whereabouts")
+        self.assertEqual(config_suite.snapshot.incidents[0].casualties, 0)
+        self.assertEqual(config_suite.snapshot.incidents[1].location, "somewhere else")
+        self.assertEqual(config_suite.snapshot.incidents[1].casualties, 1)
+
+    def test_partial_default_values(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"].pop("default")
+
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertEqual(config_suite.snapshot.tire.rim, "green")
+        self.assertEqual(config_suite.snapshot.tire.dimension, 15)
+
+    def test_invalid_defaults(self):
+        car_schema = car.build_schema()
+        raw_config = car.build_all_default_config()
+        configsuite.ConfigSuite(raw_config, car_schema)
+
+        schema_value_content = {
+            MK.Type: types.Number,
+            MK.Default: "incorrect default type",
+            MK.Required: False,
+        }
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"] = schema_value_content
+
+        with self.assertRaises(ValueError) as msg:
```

Comment:
Please call the exception wrapper something else then `msg`.

Reply:
will fix